src/misc_tools.c

```diff
@@ -305,59 +305,53 @@ int qsort_ptr_char_array_helper(const void *str1, const void *str2)
     return strcasecmp(*(const char *const *)str1, *(const char *const *)str2);
 }
 
-static const char invalid_chars[] = {'/', '\n', '\t', '\v', '\r', '\0'};
+/* List of characters we don't allow in nicks. */
+static const char invalid_nick_chars[] = {':', '/', '\0', '\a', '\b', '\f', '\n', '\r', '\t', '\v'};
+
+/* List of characters we don't allow in single-line strings. */
+static const char invalid_string_chars[] = {'\0', '\a', '\b', '\f', '\n', '\r', '\t', '\v'};
 
 /*
- * Helper function for `valid_nick()`.
- *
- * Returns true if `ch` is not in the `invalid_chars` array.
+ * Returns true if `ch` is not in the supplied `blacklist` array.
```

Comment:
I'd still keep the "invalid_chars" name. It being a (kind of) prefix (actually the above is a mixed-endian style infix or something) of the above ties them together.